### DISCUSSION TITLE
navigate synsets w/o nested list comprehensions

### DISCUSSION THREAD

**Discussion by vsraptor** [goodmami/wn#136] [Show and tell]
More detailed example with docs : 

https://gist.github.com/vsraptor/8815944d1e75947ad0c3676c0279519a

---------

here is an easy way to navigate around w/o nested list-comprehensions using pipe module :

```
from pipe import *
syns = map(lambda x : x.synsets())
sns = map(lambda x : x.senses())
ws = map(lambda x : x.words())
```

get back connected words through synsets :  words => synsets => words

`list(wn.words('book') | syns | chain | ws )`

chain is flattening iterables

```
In [33]: list(wn.words('rock') | syns | chain | ws | chain | dedup )                                                                                                         
Out[33]: [sway, shake, rock, stone, rock, rock candy, rock music, rock and roll, rock'n'roll, rock-and-roll, rock 'n' roll, tilt, sway, careen, Rock, John Rock]

In [40]: list(wn.words('rock','v') | syns | chain | sns | chain )                                                                                                            
Out[40]: [Sense('ewn-sway-v-01879288-02'), Sense('ewn-shake-v-01879288-03'), Sense('ewn-rock-v-01879288-01'), Sense('ewn-sway-v-01880021-02'), Sense('ewn-rock-v-01880021-01')]

```

----

i also have this monkey patching : 

```
def synset_repr(self):
  return re.sub(r"[\[\]']", '', str(self.lemmas()) ).replace(', ',':')
def word_repr(self):
  return str(self.lemma()) 

wn.Synset.__repr__ = synset_repr
wn.Word.__repr__ = word_repr

```

**Comment by goodmami** [comment]
Thanks, that looks useful. But note that there's no reason to do `wn.words('rock') | syns` as you can just do `wn.synsets('rock')`.

I'll close this issue as it is not a feature request or bug report.

**Comment by vsraptor** [comment]
doesnt github have some option not to create Issue every time.. i see on some repos Q/A and Discussions which are suitable for such things

**Comment by goodmami** [comment]
> doesnt github have some option not to create Issue every time.. i see on some repos Q/A and Discussions which are suitable for such things

I've opened up discussions and moved some issues there: https://github.com/goodmami/wn/discussions/
Hope this helps!
